src/core.js

```diff
@@ -1316,6 +1317,14 @@ $.extend( $.validator, {
 			return this.optional( element ) || ( value >= param[ 0 ] && value <= param[ 1 ] );
 		},
 
+		// http://jqueryvalidation.org/step-method/
+		step: function( value, element, param ) {
+
+			// Works only for number and range input types
```

Comment:
What do you think about throwing a exception, or at least log a error into the console when in debug mode, in case someone uses this feature with unsupported input types?